### ISSUE TITLE
Kiali 1.28.1: Bulb Bugs

### ISSUE THREAD

**Issue by schroedt** [kiali/kiali#3562]
Based on discussion #3557, I assume to have identified two Bugs:

1. Bulb Information Pop-Up not selectable
When enabling RBAC (see the whole configuration at #3557), bulb indicator is not selectable anymore:
![grafik](https://user-images.githubusercontent.com/6430367/103878086-3b1e5880-50d6-11eb-94be-46988970c58c.png)
at 

![grafik](https://user-images.githubusercontent.com/6430367/103878098-41acd000-50d6-11eb-9f4a-94807e308f16.png)

When you click with your mouse on the bulb, nothing happened. Before enabling RBAC, bulb was clickable and notifying why it was orange. Mouse does not change when overing over the bulb (it does for other icons at the same corner, when clickable).

2. Incorrect Bulb indication
My orange bulb indicates a missing `Istio-eGress-Gateway`. Based on my chosen profile, I decided not installing `istio-egress-gateway`, because i do not utilize it. Bulb indicates an error, which is in conflict with my decision for istio. Is there a reason, `istio-egress-gateway` is mandatory for kiali? Can the indication be disabled; display green, matching my selected `istio-profile`?

**Comment by israel-hdez** [comment]
cc @xeviknal

**Comment by xeviknal** [comment]
hi @schroedt,

starting with 2), Kiali has the settings to define which istio components you want to track it's health. Check the `external_services.istio.component_status` into the [kiali_cr.yaml doc](https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L499-L505). The istiod, ingress, and the egress are the one by default. You should add the following into your cr:

```yaml
external_services:
  istio:
    component_status:
    - app_label: istiod
      is_core: true # or false if you want kiali to show a warning instead of a failure
    - app_label: istio-ingressgateway
      is_core: true # same
```

Unfortunately, we can't automatically configure which components are installed based on the roles since the role is something not persisted in the istio configuration. It is more a config used for the istio operator to define the pieces to install.

regarding to 1), does the kiali accound have access to the istio-system namespace with rights to fetch deployments?
- Do you have other features in kiali that doesn't work well?
- Do you see errors in the log of kiali?

**Comment by schroedt** [comment]
Hi @xeviknal,

# Bullet 1
Utilized RBAC role is `kiali-admin`:

I took it from the [kiali documentation](https://kiali.io/documentation/latest/configuration/rbac/), which is referencing[ kubernetes role.yaml template](https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/kubernetes/role.yaml):


![grafik](https://user-images.githubusercontent.com/6430367/103989336-daebed00-518f-11eb-9720-5bc0b1614a59.png)


```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: kiali-admins
  labels:
    app: kiali
    version: v1.28.1
    app.kubernetes.io/name: kiali
    app.kubernetes.io/version: v1.28.1
    app.kubernetes.io/part-of: kiali
rules:
- apiGroups: [""]
  resources:
  - configmaps
  - endpoints
  - namespaces
  - nodes
  - pods
  - pods/log
  - pods/proxy
  - replicationcontrollers
  - services
  verbs:
  - get
  - list
  - patch
  - watch
- apiGroups: [""]
  resources:
  - pods/portforward
  verbs:
  - create
  - post
- apiGroups: ["extensions", "apps"]
  resources:
  - deployments
  - replicasets
  - statefulsets
  verbs:
  - get
  - list
  - patch
  - watch
- apiGroups: ["autoscaling"]
  resources:
  - horizontalpodautoscalers
  verbs:
  - get
  - list
  - watch
- apiGroups: ["batch"]
  resources:
  - cronjobs
  - jobs
  verbs:
  - get
  - list
  - patch
  - watch
- apiGroups:
  - networking.istio.io
  - security.istio.io
  resources: ["*"]
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - watch
- apiGroups: ["monitoring.kiali.io"]
  resources:
  - monitoringdashboards
  verbs:
  - get
  - list
- apiGroups: ["iter8.tools"]
  resources:
  - experiments
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - watch
```

# Bullet 2
my `kiali-cr.yaml`looked already like following:

```
    istio:
      component_status:
        enabled: true
        components:
        - app_label: istiod
          is_core: true
        - app_label: istio-ingressgateway
          is_core: true
        - app_label: istio-egressgateway
          is_core: false
      config_map_name: "istio"
      envoy_admin_local_port: 15000
      istio_identity_domain: "svc.cluster.local"
      istio_injection_annotation: "sidecar.istio.io/inject"
      istio_sidecar_annotation: "sidecar.istio.io/status"
      url_service_version: ""
```

Based on your remarks, I can set the information for `egress` to w`warning`or `error`. I would like to set it to `ignore/not installed`.

**Comment by xeviknal** [comment]
1) Correct me if I am wrong, when you use the `kiali-admins` clusterrole, the tooltip of the bulb doesn't appear. 
Which browser do you use? Does your browser show any javascript error?

2) Then you should remove the `istio-egressgateway` item from the `components` list:

```yaml
    istio:
      component_status:
        enabled: true
        components:
        - app_label: istiod
          is_core: true
        - app_label: istio-ingressgateway
          is_core: true
```

**Comment by schroedt** [comment]
### Topic 1
No Errors within log:

![grafik](https://user-images.githubusercontent.com/6430367/104033132-6a64c080-51cf-11eb-9603-1f8cdcaf7afc.png)

### Topic 2

I updated my config, let's see! :-)

**Comment by xeviknal** [comment]
hello @schroedt,

topic 1), I am not able to reproduce the problem. I am using dex, openid, rbac to allow access to only bookinfo and latest firefox (84.0.2). And it works fine.

Do you mind sharing with us the result of the following API call?
```
http://kiali-ingress:ingress-port/kiali/api/istio/status
```
I have the hunch that this is about javascript or react, not the backend.

2) Happy to see it is working!

**Comment by lucasponce** [comment]
@schroedt @xeviknal I think this issue seems resolved, just cc-ing you to update it if there is any followup task.

Feel free to close it if everything is working ok, thanks !

**Comment by schroedt** [comment]
> hello @schroedt,
> 
> topic 1), I am not able to reproduce the problem. I am using dex, openid, rbac to allow access to only bookinfo and latest firefox (84.0.2). And it works fine.
> 
> Do you mind sharing with us the result of the following API call?
> 
> ```
> http://kiali-ingress:ingress-port/kiali/api/istio/status
> ```
> 
> I have the hunch that this is about javascript or react, not the backend.
> 
>     1. Happy to see it is working!

Hi @xeviknal,

![grafik](https://user-images.githubusercontent.com/6430367/106143175-b1acf400-6172-11eb-8074-7eb8e9c3509e.png)

**Comment by schroedt** [comment]
> @schroedt @xeviknal I think this issue seems resolved, just cc-ing you to update it if there is any followup task.
> 
> Feel free to close it if everything is working ok, thanks !

Hi @lucasponce,

what needs to be done for reflection:
- Operator Update (`kiali-cr.yaml`)?
- Kiali update (`kiali.kiali.io/kiali` at `istio-system`)?

**Comment by xeviknal** [comment]
@schroedt I think that the URL to point is `/kiali/api/istio/status` instead. 
However, this would work only if you have the bulb present in the masthead. Is it still present?
In case it is not, I'd rather close the issue and if it happens ever again, feel free to open it.

**Comment by xeviknal** [comment]
@schroedt I don't fully follow your second message. Do you mind elaborating a bit more? What do you mean by reflection?

**Comment by schroedt** [comment]
![grafik](https://user-images.githubusercontent.com/6430367/106151238-70b9dd00-617c-11eb-9524-11f166959e6b.png)

**Comment by xeviknal** [comment]
@schroedt I see there is an empty array. I means that there are no problems in the istio status. If I understood it rightly, there are no problems anymore. I'll close the issue unless I am wrong and there is something we should fix in our side.
Thanks

**Comment by schroedt** [comment]
Hi @xeviknal,

I guess the bubble disappeared, once disabling `istio egress` within the config (see topic 2). I reverted my change and delete kiali pod at `istio-system`. The bulb stays hidden, whatever I configure (no istio profile having egress-gateway configured):

![grafik](https://user-images.githubusercontent.com/6430367/106243891-6b0ad880-620a-11eb-9f0c-264e5150da52.png)


**Config 1**
```
    istio:
      component_status:
        components:
        - app_label: istiod
          is_core: true
        - app_label: istio-ingressgateway
          is_core: true
```

**Config 2**
```
    istio:
      component_status:
        components:
        - app_label: istiod
          is_core: true
        - app_label: istio-ingressgateway
          is_core: true
        - app_label: istio-egressgateway
          is_core: true
```

**Config 3***
```
    istio:
      component_status:
        components:
        - app_label: istiod
          is_core: true
        - app_label: istio-ingressgateway
          is_core: true
        - app_label: istio-egressgateway
          is_core: false
```

Is this intended? Kiali operator has **not** been updated.

**Comment by schroedt** [comment]
I stayed on the page for 30 minutes now, bubble is back:
![grafik](https://user-images.githubusercontent.com/6430367/106244472-39ded800-620b-11eb-8e06-665fa25a42b0.png)

I noticed it appears via "hovering" with the mouse, satring with a few seconds delay. All other icons occur when clicking. I guess that's my issue; probably a usability issue. Any chance to show the bubbles content when clicking?

Makes no sense having one button behave differently from all other.

![grafik](https://user-images.githubusercontent.com/6430367/106244643-84f8eb00-620b-11eb-84bf-1becd8bde19e.png)
